wordpress-mobile/WordPress-Android
# [Jetpack Social] Implement CTAs in post settings screen (part 1)

Part of #18571 

This PR partially implements the CTAs in post settings screen. It's a work in progress and there will be a part 2 where the CTAS are fully functional.

To test:
**Connection list and message container disabled for published post**
The switch widget status (enabled or disabled) for a published post is still not working, since we're not yet parsing the post metadata regarding the selected accounts to share that post.

1 - Make sure `JetpackSocialFeatureConfig` feature flag is enabled;
2 - Select a site that has at least one social sharing connection;
3 - Open post settings of a published post: social sharing elements (connections and message container) should be disabled;

**Connection list and message container enabled for draft post**
Connection list switch tapping will be handled in my next PR.

1 - Make sure `JetpackSocialFeatureConfig` feature flag is enabled;
2 - Select a site that has at least one social sharing connection;
3 - Open post settings of a draft post: social sharing elements (connections and message container) should be enabled;

**"Subscribe to share more" button**
1 - Log in using a .com account;
2 - Make sure `JetpackSocialFeatureConfig` feature flag is enabled;
3 - Select a SH site that has at least one social sharing connection;
4 - Open post settings;
5 - Scroll to `"Social sharing"` section and tap the button `"Subscribe to share more"`: the WebView should be opened with the Jetpack Social Basic checkout page;

**Social sharing container gone on SH login**
1 - Log in using a SH site;
2 - Make sure `JetpackSocialFeatureConfig` feature flag is enabled;
3 - Select a site that has at least one social sharing connection;
4 - Open post settings: `"Social Sharing"` section should be gone.

**"Connect accounts" button**
1 - Remove all connected accounts in Social screen;
2 - Open posts list;
3 - Tap on a draft post;
4 - Tap the overflow menu and then "Post settings";
5 - Scroll to the "Social Sharing" section;
6 - Tap "Connect accounts" button;
7 - Select a social network to connect a new account;
8 - Tap "Connect" and "Go to web";
9 - Connect the social network account in the WebView;
10 - Move back to the post settings screen: the new connection should be shown in the "Social Sharing" section.

## Regression Notes
1. Potential unintended areas of impact
None (code behind feature flag)

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing. Unit tests will be updated/implemented in another PR.

3. What automated tests I added (or what prevented me from doing so)
Unit tests will be updated/implemented in another PR.

PR submission checklist:


UI Changes testing checklist:
